File: src/schemas/analytics.ts
export type DailySalesSummary = {
  sale_date: string
  day_of_week: string
  total_sales_count: number
  total_sales_value: number
  avg_sale_value: number
}

export type TopProductTrend = {
  period: string
  product_id: number
  product_name: string
  total_units_sold: number
  total_revenue: number
}

export type InventorySnapshot = {
  product_id: number
  name: string
  sku: string
  category: string
  stock_quantity: number
}

export type InventoryMovement = {
  product_id: number
  name: string
  category: string
  total_added: number
  total_removed: number
  net_change: number
}

export type TopMarginProduct = {
  product_id: number
  name: string
  category: string
  units_sold: number
  revenue: number
  cost_basis: number
  margin_dollars: number
  margin_percent: number
}

export type CategorySales = {
  category: string
  total_units_sold: number
  total_revenue: number
  avg_price: number
}

export type LowStockAlert = {
  product_id: number
  name: string
  category: string
  stock_quantity: number
}

// Return analytics
export type ProductReturnRate = {
  product_id: number
  product_name: string
  total_sold: number
  total_returned: number
  return_rate: number
}

export type SaleReturnRate = {
  sale_id: number
  total_items_sold: number
  total_items_returned: number
  return_rate: number
}

export type CashierReturnRate = {
  user_id: number
  username: string
  total_returns: number
  total_sales: number
  return_ratio: number
}

File: schemas/analytics.py
# schemas/analytics.py

from pydantic import BaseModel
from typing import Optional


class DailySalesSummary(BaseModel):
    sale_date: str
    day_of_week: str
    total_sales_count: int
    total_sales_value: float
    avg_sale_value: float


class TopProductTrend(BaseModel):
    sale_date: str
    product_id: int
    product_name: str
    units_sold: int
    revenue: float


class InventorySnapshot(BaseModel):
    id: int
    name: str
    category: str
    quantity: int
    price: float


class InventoryMovement(BaseModel):
    product_id: int
    name: str
    category: str
    total_added: int
    total_removed: int
    net_change: int


class TopMarginProduct(BaseModel):
    product_id: int
    name: str
    category: str
    units_sold: int
    revenue: float
    cost_basis: float
    margin_dollars: float
    margin_percent: float


class CategorySales(BaseModel):
    category: str
    total_units_sold: int
    total_revenue: float
    avg_price: float


class KpiSummary(BaseModel):
    total_sales: float
    avg_daily_sales: float
    total_transactions: int
    avg_basket_size: float
    return_rate: float
    top_category: str

    # Add deltas
    delta_total_sales: Optional[float] = None
    delta_avg_daily_sales: Optional[float] = None
    delta_total_transactions: Optional[float] = None
    delta_avg_basket_size: Optional[float] = None
    delta_return_rate: Optional[float] = None

    model_config = {"from_attributes": True}
